### PR TITLE
Clean up some navigation tests

### PR DESCRIPTION
**Pull request by domenic**
- Rewrite several navigation-during-unload tests to use fewer auxiliary files and be readable all inline

- Update about-srcdoc-navigation-blocked.html to use some of the common helpers for better readability.

- Remove 002.html and its supporting files since they were redundant with the faster overlapping-navigations-and-traversals/cross-document-nav-cross-document-nav.html that was recently introduced.

### CODE REVIEW

File: html/browsers/browsing-the-web/navigating-across-documents/about-srcdoc-navigation-blocked.window.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  await waitForIframeLoad(iframe);
+
+  iframe.contentWindow.location = "about:srcdoc";
+
+  // Fetching "about:srcdoc" should result in a network error, and navigating
+  // to a network error should produce an opaque-origin page. In particular,
+  // since the error page should end up being cross-origin to the parent
+  // frame, `contentDocument` should return `null`.
+  //
+  // If instead this results in a message because we reloaded the srcdoc iframe, immediately fail.
```
Comment (domfarolino) [inline html/browsers/browsing-the-web/navigating-across-documents/about-srcdoc-navigation-blocked.window.js:17]: I guess this is a bug we're seeing browsers, where the non-about:srcdoc page actually gets reloaded (and the parent frame gets a message)?

**Review by domfarolino — APPROVED** [review]
Looks great, thanks!

While we're here, would you mind adding a test similar to `html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-data-url.window.js` but for javascript: URLs too, since it looks like they should also be ignored?

File: html/browsers/browsing-the-web/navigating-across-documents/about-srcdoc-navigation-blocked.window.js (hunk 1 above)
Comment (domenic) [reply html/browsers/browsing-the-web/navigating-across-documents/about-srcdoc-navigation-blocked.window.js:17]: I think the bug involves navigating to the srcdoc document that was created on line 5. I will tweak the wording to not use "reload".

**Review by domenic — COMMENTED** [review]
> While we're here, would you mind adding a test similar to html/browsers/browsing-the-web/navigating-across-documents/navigation-unload-data-url.window.js but for javascript: URLs too, since it looks like they should also be ignored?

Will do!

**Pull request merged**